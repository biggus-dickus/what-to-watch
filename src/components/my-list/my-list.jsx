import React from 'react';
import PropTypes from 'prop-types';

import Catalog from '../catalog/catalog';
import Footer from '../partials/footer/footer';
import Logo from '../partials/logo/logo';
import UserBlock from '../partials/user-block/user-block';


const MyList = (props) => {
  const {location, userData, selectedMovies = []} = props;

  return (
    <div className="user-page">
      <header className="page-header user-page__head">
        <Logo pathname={props.location.pathname} />

        <h1 className="page-title user-page__title">My list</h1>

        <UserBlock location={props.location} user={userData} />
      </header>

      <Catalog movies={selectedMovies} />

      <Footer pathname={location.pathname} />
    </div>
  );
};

MyList.propTypes = {
  location: PropTypes.object.isRequired,
  userData: PropTypes.object,
  selectedMovies: PropTypes.arrayOf(PropTypes.object)
};

export default MyList;
