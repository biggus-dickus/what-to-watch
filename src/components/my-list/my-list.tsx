import * as React from 'react';

import Catalog from '../catalog/catalog';
import Footer from '../partials/footer/footer';
import Logo from '../partials/logo/logo';
import UserBlock from '../partials/user-block/user-block';

import {Film, Location, User} from '../../types'; // eslint-disable-line

interface Props {
  location: Location,
  userData: User,
  selectedMovies?: Array<Film>
}


const MyList = (props: Props): React.ReactElement => {
  const {location, userData, selectedMovies = []} = props;

  return (
    <div className="user-page">
      <header className="page-header user-page__head">
        <Logo pathname={props.location.pathname} />

        <h1 className="page-title user-page__title">My list</h1>

        <UserBlock location={props.location} user={userData} />
      </header>

      <Catalog movies={selectedMovies} />

      <Footer pathname={location.pathname} />
    </div>
  );
};

export default MyList;
