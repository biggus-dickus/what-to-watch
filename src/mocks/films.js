export const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    picUrl: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Bohemian Rhapsody`,
    picUrl: `img/bohemian-rhapsody.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Macbeth`,
    picUrl: `img/macbeth.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Aviator`,
    picUrl: `img/aviator.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `We need to talk about Kevin`,
    picUrl: `img/we-need-to-talk-about-kevin.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `What We Do in the Shadows`,
    picUrl: `img/what-we-do-in-the-shadows.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Revenant`,
    picUrl: `img/revenant.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Johny English`,
    picUrl: `img/johnny-english.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Shutter Island`,
    picUrl: `img/shutter-island.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Pulp Fiction`,
    picUrl: `img/pulp-fiction.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `No Country for Old Men`,
    picUrl: `img/no-country-for-old-men.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Snatch`,
    picUrl: `img/snatch.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Moonrise Kingdom`,
    picUrl: `img/moonrise-kingdom.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Seven Years in Tibet`,
    picUrl: `img/seven-years-in-tibet.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Midnight Special`,
    picUrl: `img/midnight-special.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `War of the Worlds`,
    picUrl: `img/war-of-the-worlds.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Dardjeeling Limited`,
    picUrl: `img/dardjeeling-limited.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Orlando`,
    picUrl: `img/orlando.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Mindhunter`,
    picUrl: `img/mindhunter.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
];
