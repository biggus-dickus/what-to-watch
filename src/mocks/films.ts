import {Film} from '../types'; // eslint-disable-line
import {Genre} from './genres';

export const mockFilms: Film[] = [
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    id: 1,
    genre: Genre.FAMILY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    bgColor: `green`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 7.77,
    released: 2019,
    runTime: 120,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Bohemian Rhapsody`,
    id: 2,
    genre: Genre.DOCUMENTARY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/bohemian-rhapsody.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    bgColor: `blue`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1654,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Macbeth`,
    id: 3,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/macbeth.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4%2FCam-Man.mp4`
  },
  {
    bgColor: `rebeccapurple`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 5.55,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Aviator`,
    id: 4,
    genre: Genre.ROMANCE,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/aviator.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/5/5b/Marnie_%281964%29_trailer.webm/Marnie_%281964%29_trailer.webm.360p.vp9.webm`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `We need to talk about Kevin`,
    id: 5,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/we-need-to-talk-about-kevin.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/5/5b/Marnie_%281964%29_trailer.webm/Marnie_%281964%29_trailer.webm.360p.vp9.webm`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `What We Do in the Shadows`,
    id: 6,
    genre: Genre.DOCUMENTARY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/what-we-do-in-the-shadows.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/2/23/Dr_Strangelove_-_Official_Trailer.webm/Dr_Strangelove_-_Official_Trailer.webm.480p.vp9.webm`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Revenant`,
    id: 7,
    genre: Genre.THRILLERS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/revenant.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Harvesting%20Rice.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: true,
    posterImage: `poster.jpg`,
    rating: 8.10,
    released: 2005,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Johny English`,
    id: 8,
    genre: Genre.COMEDIES,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/johnny-english.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Lake%20Day.mp4`
  },
  {
    bgColor: `pink`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 2009,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Shutter Island`,
    id: 9,
    genre: Genre.THRILLERS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/shutter-island.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Deserted%20Island.mp4`
  },
  {
    bgColor: `purple`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Quentin Tarantino`,
    isFavourite: true,
    posterImage: `poster.jpg`,
    rating: 9.50,
    released: 1994,
    runTime: 180,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Pulp Fiction`,
    id: 10,
    genre: Genre.CRIME,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/pulp-fiction.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Stunning%20Mountains.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `No Country for Old Men`,
    id: 11,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/no-country-for-old-men.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Gubat%20Beach.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Guy Ritchie`,
    isFavourite: true,
    posterImage: `poster.jpg`,
    rating: 8.36,
    released: 2004,
    runTime: 125,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Snatch`,
    id: 12,
    genre: Genre.COMEDIES,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/snatch.jpg`,
    previewVideo: `etc/na-zdorovnya.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Moonrise Kingdom`,
    id: 13,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/moonrise-kingdom.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Caramoan%20Drone.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Seven Years in Tibet`,
    id: 14,
    genre: Genre.DOCUMENTARY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/seven-years-in-tibet.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Ants%20Walking.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Midnight Special`,
    id: 15,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/midnight-special.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Boat%20in%20Caramoan.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `War of the Worlds`,
    id: 16,
    genre: Genre.SCI_FI,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/war-of-the-worlds.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Blue%20Joy.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Dardjeeling Limited`,
    id: 17,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/dardjeeling-limited.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Stop-Sign.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Orlando`,
    id: 18,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/orlando.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Steaming-Yellowstone.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Mindhunter`,
    id: 19,
    genre: Genre.THRILLERS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/mindhunter.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/California-Desert.mp4`
  }
];
