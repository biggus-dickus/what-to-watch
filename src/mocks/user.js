const mockUser = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatar_url: `img/1.png` // eslint-disable-line
};

export default mockUser;
