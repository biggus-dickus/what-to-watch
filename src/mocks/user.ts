import {Location, User} from '../types'; // eslint-disable-line

export const mockLocation: Location = {
  hash: `#p1t00h`,
  key: `0mfg666`,
  pathname: `/`,
  search: ``
};

const mockUser: User = {
  avatar_url: '/wtw/static/avatar/10.jpg', // eslint-disable-line
  id: 777,
  email: `test@test.com`,
  name: `Nastasha Romanenko`,
};

export default mockUser;
