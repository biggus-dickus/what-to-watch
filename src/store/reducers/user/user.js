import {ActionType} from '../../action-types';

const initialState = {
  isAuthorizationRequired: false,
  userData: null
};

export const userReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case ActionType.AUTH_REQUIRED:
      return {
        ...state,
        isAuthorizationRequired: action.payload
      };

    case ActionType.GET_USER_DATA:
      return {
        ...state,
        userData: action.payload
      };

    default: return state;
  }
};
