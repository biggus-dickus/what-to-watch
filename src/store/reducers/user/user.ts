import {ActionType} from '../../action-types';

import {GenericAction} from '../../actions'; // eslint-disable-line
import {User} from '../../../types'; // eslint-disable-line

interface State {
  authError?: null | string,
  isAuthorizationRequired: boolean,
  userData?: User | null
}

const initialState = {
  authError: null,
  isAuthorizationRequired: false,
  userData: null
};

export const userReducer = (state: State = initialState, action: GenericAction) => {
  switch (action.type) {
    case ActionType.AUTH_REQUIRED:
      return {
        ...state,
        isAuthorizationRequired: action.payload
      };

    case ActionType.GET_USER_DATA:
      return {
        ...state,
        userData: action.payload
      };

    case ActionType.AUTH_FAIL:
      return {
        ...state,
        authError: action.payload
      };

    default: return state;
  }
};
