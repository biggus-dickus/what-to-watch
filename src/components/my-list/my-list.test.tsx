import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {StaticRouter as Router} from 'react-router-dom';
import * as Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';

import {mockFilms} from '../../mocks/films';
import {mockLocation} from '../../mocks/user';
import mockUser from '../../mocks/user';

import {MyList} from './my-list';

configure({adapter: new Adapter()});

const props = {
  location: mockLocation,
  onGetWatchList: jest.fn(),
  userData: mockUser,
  watchList: mockFilms.filter((film) => film.isFavourite)
};

describe(`MyList test suite`, () => {
  it(`<MyList /> should render correctly`, () => {
    const tree = renderer.create(
        <Router>
          <MyList {...props} />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should display a notification if there is nothing in the watch list`, () => {
    const newProps = {...props, watchList: []};

    const wrapper = shallow(<MyList {...newProps} />);
    expect(wrapper.find(`[data-test="at-no-films-notification"]`)).toHaveLength(1);
  });
});
