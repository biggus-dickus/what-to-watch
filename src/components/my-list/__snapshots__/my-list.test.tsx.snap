// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyList /> should render correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <span
        className="logo__link"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </span>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="user-block__avatar"
        href="/mylist"
        onClick={[Function]}
      >
        <img
          alt="Nastasha Romanenko"
          height="63"
          src="https://es31-server.appspot.com/wtw/static/avatar/10.jpg"
          width="63"
        />
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        tabIndex={0}
      >
        <a
          className="small-movie-card__wrapping-link"
          href="/https://thepiratebay.org"
          onClick={[Function]}
        >
          <i
            aria-hidden={true}
            className="small-movie-card__play-btn"
          />
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/johnny-english.jpg"
              src="https://storage.googleapis.com/coverr-main/mp4/Lake%20Day.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <span
              className="small-movie-card__link"
            >
              Johny English
            </span>
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        tabIndex={0}
      >
        <a
          className="small-movie-card__wrapping-link"
          href="/https://thepiratebay.org"
          onClick={[Function]}
        >
          <i
            aria-hidden={true}
            className="small-movie-card__play-btn"
          />
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/pulp-fiction.jpg"
              src="https://storage.googleapis.com/coverr-main/mp4/Stunning%20Mountains.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <span
              className="small-movie-card__link"
            >
              Pulp Fiction
            </span>
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        tabIndex={0}
      >
        <a
          className="small-movie-card__wrapping-link"
          href="/https://thepiratebay.org"
          onClick={[Function]}
        >
          <i
            aria-hidden={true}
            className="small-movie-card__play-btn"
          />
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/snatch.jpg"
              src="etc/na-zdorovnya.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <span
              className="small-movie-card__link"
            >
              Snatch
            </span>
          </h3>
        </a>
      </article>
    </div>
    <div
      className="catalog__more"
    >
      <button
        className="catalog__button"
        type="button"
      >
        Show more
      </button>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <span
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </span>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 
        2019
         What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
